spring-cloud-data-core/src/main/java/org/springframework/cloud/data/core/StreamDefinition.java

```diff
@@ -126,6 +147,57 @@ public String toString() {
 
 
 	/**
+	 * Build a list of ModuleDefinitions out of a parsed StreamNode.
+	 *
+	 * @param name the name of the definition unit
+	 * @param stream the AST construct representing the definition
+	 */
+	private static List<ModuleDefinition> buildModuleDefinitions(String name, StreamNode stream) {
```

Comment:
There's a lot going on in here for a private static method. Perhaps this should go into a builder of some sort?